File: ios/bittr/Map/PlaceCategories.swift
//
//  PlaceCategories.swift
//  bittr
//
//  Created by Tom Melters on 4/27/26.
//

import Foundation

extension String? {
    
    func iconName() -> String {
        
        switch self {
            
        // Food & Drink
        case "local_cafe":
            return "cup.and.saucer.fill"
        case "restaurant", "local_dining", "lunch_dining":
            return "fork.knife"
        case "local_pizza":
            return "fork.knife.circle.fill"
        case "local_bar", "bar", "sports_bar":
            return "wineglass.fill"
        case "cake":
            return "birthday.cake.fill"
        case "icecream":
            return "birthday.cake"
            
        // Stay / Accommodation
        case "hotel":
            return "bed.double.fill"
        case "cottage":
            return "house.fill"
            
        // Bitcoin / Finance
        case "local_atm":
            return "bitcoinsign.circle.fill"
        case "account_balance":
            return "building.columns.fill"
            
        // Shopping / Retail
        case "shopping_cart", "store", "supermarket":
            return "cart.fill"
        case "storefront":
            return "storefront.fill"
        case "local_grocery_store":
            return "basket.fill"
        case "card_giftcard":
            return "gift.fill"
        case "business":
            return "building.2.fill"
            
        // Health / Beauty
        case "spa":
            return "leaf.fill"
        case "medical_services":
            return "cross.case.fill"
        case "fitness_center":
            return "figure.strengthtraining.traditional"
        case "content_cut":
            return "scissors"
            
        // Vehicles / Transport
        case "directions_car", "car_repair":
            return "car.fill"
        case "commute":
            return "tram.fill"
        case "luggage":
            return "suitcase.fill"
            
        // Tech / Electronics
        case "computer":
            return "desktopcomputer"
        case "smartphone":
            return "iphone"
            
        // Outdoor / Leisure
        case "pedal_bike":
            return "bicycle"
        case "pool":
            return "figure.pool.swim"
        case "grass":
            return "leaf.arrow.circlepath"
            
        // Education / Work
        case "school":
            return "graduationcap.fill"
        case "group":
            return "person.3.fill"
        case "engineering":
            return "wrench.and.screwdriver.fill"
        case "design_services":
            return "paintbrush.fill"
        case "palette":
            return "paintpalette.fill"
            
        // Misc
        case "visibility":
            return "eye.fill"
        case "question_mark":
            return "questionmark.circle.fill"
        case "liquor":
            return "wineglass"
        case "sports_soccer":
            return "soccerball"
        case "vaping_rooms":
            return "smoke.fill"
            
        default:
            return "mappin.circle.fill"
        }
    }
        
    func categoryDescription() -> String {
        
        switch self {
            
        // Food & Drink
        case "local_cafe":
            return "Café"
        case "restaurant", "local_dining":
            return "Restaurant"
        case "lunch_dining":
            return "Eatery"
        case "local_pizza":
            return "Pizza place"
        case "local_bar":
            return "Bar"
        case "sports_bar":
            return "Sports bar"
        case "cake":
            return "Bakery"
        case "icecream":
            return "Ice cream shop"
        case "liquor":
            return "Liquor store"
            
        // Stay
        case "hotel":
            return "Hotel"
        case "cottage":
            return "Accommodation"
            
        // Bitcoin / Finance
        case "local_atm":
            return "Bitcoin ATM"
        case "account_balance":
            return "Financial services"
            
        // Retail
        case "shopping_cart", "store", "supermarket":
            return "Shop"
        case "storefront":
            return "Store"
        case "local_grocery_store":
            return "Grocery store"
        case "card_giftcard":
            return "Gift shop"
        case "business":
            return "Business"
            
        // Health / Beauty
        case "spa":
            return "Spa"
        case "medical_services":
            return "Medical services"
        case "fitness_center":
            return "Gym"
        case "content_cut":
            return "Hairdresser"
            
        // Vehicles / Transport
        case "directions_car":
            return "Car dealer"
        case "car_repair":
            return "Car repair"
        case "commute":
            return "Transport"
        case "luggage":
            return "Travel services"
            
        // Tech
        case "computer":
            return "Computer store"
        case "smartphone":
            return "Phone shop"
            
        // Outdoor / Leisure
        case "pedal_bike":
            return "Bike shop"
        case "pool":
            return "Watersports"
        case "grass":
            return "Garden services"
            
        // Education / Work
        case "school":
            return "School"
        case "group":
            return "Community"
        case "engineering":
            return "Engineering"
        case "design_services":
            return "Design services"
        case "palette":
            return "Art"
            
        // Misc
        case "visibility":
            return "Optical services"
        case "question_mark":
            return "Business"
        case "sports_soccer":
            return "Sports club"
        case "vaping_rooms":
            return "Vape shop"
            
        default:
            return "Business"
        }
    }
}

File: android/feature/map/src/main/kotlin/com/bittr/android/feature/map/PlaceCategories.kt
package com.bittr.android.feature.map

/**
 * BTCMap's icon slug → the words the place sheet shows.
 *
 * Ported from `categoryDescription()` in `ios/bittr/Map/PlaceCategories.swift`,
 * entry for entry and with the same default: an unrecognised slug reads "Business"
 * rather than the raw slug, so a new BTCMap category shows something sensible
 * instead of `vaping_rooms`.
 *
 * The sibling `iconName()` is deliberately **not** ported: it maps to SF Symbols,
 * which do not exist here, and inventing an Android icon set for 40 categories is
 * design work rather than transcription. The sheet shows the words until it has one.
 */
internal fun categoryDescription(icon: String?): String = when (icon) {
    // Food & drink
    "local_cafe" -> "Café"
    "restaurant", "local_dining" -> "Restaurant"
    "lunch_dining" -> "Eatery"
    "local_pizza" -> "Pizza place"
    "local_bar" -> "Bar"
    "sports_bar" -> "Sports bar"
    "cake" -> "Bakery"
    "icecream" -> "Ice cream shop"
    "liquor" -> "Liquor store"

    // Stay
    "hotel" -> "Hotel"
    "cottage" -> "Accommodation"

    // Bitcoin / finance
    "local_atm" -> "Bitcoin ATM"
    "account_balance" -> "Financial services"

    // Retail
    "shopping_cart", "store", "supermarket" -> "Shop"
    "storefront" -> "Store"
    "local_grocery_store" -> "Grocery store"
    "card_giftcard" -> "Gift shop"
    "business" -> "Business"

    // Health & beauty
    "spa" -> "Spa"
    "medical_services" -> "Medical services"
    "fitness_center" -> "Gym"
    "content_cut" -> "Hairdresser"

    // Vehicles & transport
    "directions_car" -> "Car dealer"
    "car_repair" -> "Car repair"
    "commute" -> "Transport"
    "luggage" -> "Travel services"

    // Tech
    "computer" -> "Computer store"
    "smartphone" -> "Phone shop"

    // Outdoor & leisure
    "pedal_bike" -> "Bike shop"
    "pool" -> "Watersports"
    "grass" -> "Garden services"

    // Education & work
    "school" -> "School"
    "group" -> "Community"
    "engineering" -> "Engineering"
    "design_services" -> "Design services"
    "palette" -> "Art"

    // Misc
    "visibility" -> "Optical services"
    "question_mark" -> "Business"
    "sports_soccer" -> "Sports club"
    "vaping_rooms" -> "Vape shop"

    else -> "Business"
}

/**
 * The website as the sheet shows it: scheme and `www.` stripped, no trailing slash.
 *
 * Ported from `showWebsite` (`OnePlaceViewController.swift:79-89`). The stripping is
 * display only — the in-app browser is handed the original string, because
 * `example.com` with no scheme is not a URL a `WebView` will load.
 */
internal fun displayWebsite(website: String): String =
    website
        .removePrefix("https://")
        .removePrefix("http://")
        .removePrefix("www.")
        .removeSuffix("/")
